Ignore charset decoding errors when parsing MIME messages

marinthiercelin opened the pull request:
Instead of assuming utf8 when no charset is provided,
DecodeCharset will return the text as is.
This makes the parser more permissive on what messages
can be considered valid.

twiss commented on encoding.go:
```diff
@@ -184,10 +185,7 @@ func DecodeCharset(original []byte, mediaType string, contentTypeParams map[stri
-			return original, nil
-		}
-		err = fmt.Errorf("non-utf8 content without charset specification")
+		return original, nil
```
With this change, we should also update the comment above the function

Commit:
Ignore charset decoding error when parsing MIME messages.

Instead of failing the entire parsing when unable to decode the charset,
we continue the parsing with the original text as is.
We log the error as a warning before continuing.
